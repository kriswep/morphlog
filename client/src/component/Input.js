import React from 'react';
import styled from 'styled-components';

const Container = styled.div`
  display: flex;
  flex-direction: column;
`;

const Label = styled.label`
  line-height: 1rem;
  padding: 0.1rem 0;
  font-size: 0.8rem;
`;

const Input = styled.input`
  background-color: transparent;
  border: 1px solid ${props => props.theme.lightAccent};
  padding: 0.5rem 0.25rem;
  box-shadow: -0.05rem 0.05rem 0.2rem ${props => props.theme.lightAccent};
  border-radius: 0.2rem;
  &:focus {
    outline-color: ${props => props.theme.lightAccent};
  }
  &::placeholder {
    color: ${props => props.theme.lightAccent};
  }
`;

const Textarea = Input.withComponent('textarea').extend`
  resize: none;
  height: 8rem;
`;

const MyInput = ({ textarea, name, label, ...rest }) => {
  return (
    <Container>
      {name && label && <Label for={name}>{label}</Label>}
      {textarea ? (
        <Textarea name={name} {...rest} />
      ) : (
        <Input name={name} {...rest} />
      )}
    </Container>
  );
};

export default MyInput;
