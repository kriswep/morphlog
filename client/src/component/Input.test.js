/* globals test expect */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Input from './Input';

test('Input renders a input', () => {
  const wrapper = shallow(<Input name="myName" data-prop="myData" />);
  expect(wrapper).toExist();
  expect(toJson(wrapper)).toMatchSnapshot();
});

test('Input renders a textarea', () => {
  const wrapper = shallow(
    <Input textarea label="myLabel" name="myName" data-prop="myData" />,
  );
  expect(wrapper).toExist();
  expect(toJson(wrapper)).toMatchSnapshot();
});
