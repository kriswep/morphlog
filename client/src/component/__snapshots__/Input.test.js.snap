// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders a input 1`] = `
<styled.div>
  <styled.input
    data-prop="myData"
    name="myName"
  />
</styled.div>
`;

exports[`Input renders a textarea 1`] = `
<styled.div>
  <styled.label
    for="myName"
  >
    myLabel
  </styled.label>
  <styled.textarea
    data-prop="myData"
    name="myName"
  />
</styled.div>
`;
